drivers/sensor/apds9960/apds9960.c

```diff
@@ -0,0 +1,489 @@
+		val->val2 = 0;
+		break;
+	case SENSOR_CHAN_GREEN:
+		val->val1 = data->sample_g;
+		val->val2 = 0;
+		break;
+	case SENSOR_CHAN_BLUE:
+		val->val1 = data->sample_b;
+		val->val2 = 0;
+		break;
+	case SENSOR_CHAN_PROX:
+		val->val1 = data->pdata;
+		val->val2 = 0;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int apds9960_setproxint_lowthresh(struct device *dev, u8_t threshold)
+{
+	struct apds9960_data *data = dev->driver_data;
+	int temp = 0;
+
+	temp = i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_PILT_REG, threshold);
+	if (temp < 0) {
+		SYS_LOG_DBG(" Failed");
+		return temp;
+	}
+
+	return 0;
+}
+
+static int apds9960_setproxint_highthresh(struct device *dev, u8_t threshold)
+{
+	struct apds9960_data *data = dev->driver_data;
+	int temp = 0;
+
+	temp = i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_PIHT_REG, threshold);
+	if (temp < 0) {
+		SYS_LOG_DBG(" Failed");
+		return temp;
+	}
+
+	return 0;
+}
+
+static int apds9960_setlightint_lowthresh(struct device *dev, u16_t threshold)
+{
+	struct apds9960_data *data = dev->driver_data;
+	int temp = 0;
+	u8_t val_low;
+	u8_t val_high;
+
+	/* Break 16-bit threshold into 2 8-bit values */
+	val_low = threshold & 0x00FF;
+	val_high = (threshold & 0xFF00) >> 8;
+
+	temp = i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_INT_AILTL_REG, val_low);
+	if (temp < 0) {
+		SYS_LOG_DBG(" Failed");
+		return temp;
+	}
+
+	temp = i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_INT_AILTH_REG, val_high);
+	if (temp < 0) {
+		SYS_LOG_DBG(" Failed");
+		return temp;
+	}
+
+	return 0;
+}
+
+static int apds9960_setlightint_highthresh(struct device *dev, u16_t threshold)
+{
+	struct apds9960_data *data = dev->driver_data;
+	int temp = 0;
+	u8_t val_low;
+	u8_t val_high;
+
+	/* Break 16-bit threshold into 2 8-bit values */
+	val_low = threshold & 0x00FF;
+	val_high = (threshold & 0xFF00) >> 8;
+
+	temp = i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_INT_AIHTL_REG, val_low);
+	if (temp < 0) {
+		SYS_LOG_DBG(" Failed");
+		return temp;
+	}
+
+	temp = i2c_reg_write_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_INT_AIHTH_REG, val_high);
+	if (temp < 0) {
+		SYS_LOG_DBG(" Failed");
+		return temp;
+	}
+
+	return 0;
+}
+
+static int apds9960_proxy_setup(struct device *dev, int gain)
+{
+	struct apds9960_data *data = dev->driver_data;
+	int temp = 0;
+
+	/* Power ON */
+	temp = i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ENABLE_REG, BIT(0), APDS9960_POWER_ON);
+	if (temp < 0) {
+		SYS_LOG_DBG("Power on bit not set.");
+		return temp;
+	}
+
+	/* ADC value */
+	temp = i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_ATIME_REG, APDS9960_ATIME_WRTIE, APDS9960_ADC_VALUE);
+	if (temp < 0) {
+		SYS_LOG_DBG("ADC bits are not written");
+		return temp;
+	}
+
+	/* proxy Gain */
+	temp = i2c_reg_update_byte(data->i2c, APDS9960_I2C_ADDRESS,
+		APDS9960_CONTROL_REG, APDS9960_BITS_32,
```

Comment:
APDS9960_BITS_32 needs a more descriptive name, such as APDS9960_CONTROL_PGAIN. Same for the other bitfields